pkg/datapath/iptables/iptables.go

```diff
@@ -744,15 +764,15 @@ func InstallNoTrackRules(IP string, port uint16, ipv6 bool) error {
 	}
 	ports := noTrackPorts(port)
 	for _, p := range ports {
-		if err := noTrackRules(prog, "-A", IP, p, true); err != nil {
+		if err := endpointNoTrackRules(prog, "-A", IP, p, true); err != nil {
 			log.WithFields(logrus.Fields{
 				ipField:            IP,
 				logfields.Port:     p.Port,
 				logfields.Protocol: p.Protocol,
 			}).WithError(err).Warn("Unable to install ingress NOTRACK rules")
 			return err
 		}
-		if err := noTrackRules(prog, "-A", IP, p, false); err != nil {
+		if err := endpointNoTrackRules(prog, "-A", IP, p, false); err != nil {
```

Comment:
This is outside of the scope of this PR, but I wonder if we need to call `RemoveEndpointNoTrackRules` (or do something similar) to remove the previous rule that was successfully inserted here.

Reply:
Unless I'm missing something I don't think it's possible to have per-endpoint no CT rules in place when the `--install-no-conntrack-iptables-rules` option is set :thinking: (as it's not something you can change after starting the agent)

Reply:
Ah -- I meant if a previous call (e.g., the one on line 767) succeeds but this fails.